pytorch_lightning/utilities/auto_restart.py

```diff
@@ -143,8 +144,10 @@ def __init__(self, dataset: IterableDataset) -> None:
     def sampler(self) -> Sampler:
         return self.dataset.sampler
 
-    def state_dict(self) -> Dict[str, Any]:
-        return {k: v.state_dict() for k, v in self.samplers.items()}
+    def state_dict(self) -> Optional[Dict[str, Any]]:
+        if isinstance(self.samplers, dict):
+            return {k: v.state_dict() for k, v in self.samplers.items()}
+        return None
```

Comment:
I think this should assert that self.samplers is a dict or return an empty dict, not None